File: Java/DecoratorPattern/src/com/laxmena/beverage/condiments/Mocha.java
package com.laxmena.beverage.condiments;

import com.laxmena.beverage.Beverage;

public class Mocha extends CondimentDecorator {

    public Mocha(Beverage beverage) {
        this.beverage = beverage;
    }
    @Override
    public double cost() {
        return this.beverage.cost() + 0.20;
    }

    @Override
    public String getDescription() {
        return this.beverage.getDescription() + ", Mocha";
    }
}

File: Decorator/Python/Coffee-Hut/beverage/condiments/mocha.py
from beverage.beverage import Beverage
from beverage.condiments.condiments_decorator import CondimentsDecorator


class Mocha(CondimentsDecorator):
    def __init__(self, beverage: Beverage):
        self.beverage = beverage

    def get_description(self) -> str:
        return self.beverage.get_description() + ", Mocha"

    def cost(self) -> float:
        return self.beverage.cost() + 0.20
